File: leetcode/3740.cpp
class Solution {
  public:
    int minimumDistance(vector<int>& nums) {
        int n = nums.size();
        int base = 200;
        int min_distance = base;

        for (int i = 0; i < (n - 2); i++) {
            for (int j = i + 1; j < (n - 1); j++) {
                for (int k = j + 1; k < n; k++) {
                    if (nums[i] == nums[j] && nums[j] == nums[k]) {
                        min_distance = min(min_distance, (k - i) << 1);
                    }
                }
            }
        }

        return (min_distance == base) ? -1 : min_distance;
    }
};

File: leetcode/3740.py
class Solution:
    def minimumDistance(self, nums: List[int]) -> int:
        n = len(nums)
        default = 2**31 - 1
        min_distance = default
        def calculate_distance(a, b, c):
            return (b - a) + (c - b) + (c - a)
        
        for i in range(n-2):
            for j in range(i+1, n-1):
                for k in range(j+1, n):
                    if nums[i] == nums[j] == nums[k]:
                        min_distance = min(min_distance, calculate_distance(i, j, k))

        return -1 if min_distance == default else min_distance
